src/engagement_db_to_coda/engagement_db_to_coda.py

```diff
@@ -155,10 +154,40 @@ def _sync_message_to_coda(transaction, engagement_db, coda, coda_config, message
     :type engagement_db: engagement_database.EngagementDatabase
     :param coda: Coda instance to sync the message to.
     :type coda: coda_v2_python_client.firebase_client_wrapper.CodaV2Client
-    :param message_id: Id of the message to sync.
-    :type message_id: str
+    :param coda_config: Coda sync configuration.
+    :type coda_config: src.engagement_db_to_coda.configuration.CodaSyncConfiguration
+    :param last_seen_message: Last seen message, downloaded from the database in a previous call, or None.
+                              If provided, downloads the least recently updated (next) message after this one, otherwise
+                              downloads the least recently updated message in the database.
+    :type last_seen_message: engagement_database.data_models.Message | None
+    :return: The engagement database message that was synced.
+             If there was no new message to sync, returns None.
+    :rtype: engagement_database.data_models.Message | None
     """
-    engagement_db_message = engagement_db.get_message(message_id, transaction=transaction)
+    if last_seen_message is None:
+        messages_filter = lambda q: q \
+            .where("status", "in", [MessageStatuses.LIVE, MessageStatuses.STALE]) \
```

Comment:
do we expect a message without any of these two status?

Reply:
yes, the other 2 statuses we discussed were archived and anonymised, and we don't want to analyse messages with these statuses so no need to sync to coda